scripts/speech_recognition/estimate_duration_bins.py

```diff
@@ -0,0 +1,100 @@
+# Copyright (c) 2024, NVIDIA CORPORATION.  All rights reserved.
```

Comment:
How about calling these functions from inside the tarred dataset builder script by default - ie always have the defaults ready so user doesn't forget.

Reply:
Good idea, I added it. The tarred conversion script will now output some extra info in the config and console:

```
$ python scripts/speech_recognition/convert_to_tarred_audio_dataset.py --manifest_path manifest.json --num_shards 4 --workers 4 --max_duration 30 --dynamic_buckets_num 10

/Users/pzelasko/code/nemo/scripts/speech_recognition/create_dali_tarred_dataset_index.py:71: UserWarning: 
The version_base parameter is not specified.
Please specify a compatability version level, or None.
Will assume defaults for version 1.1
  @hydra.main(config_path=None, config_name='index_config')
Creating new tarred dataset ...
After filtering, manifest has 1519 files which amounts to 19118.67481250002 seconds of audio.
Number of samples added : 1519
Remainder: 3
Shard 0 has entries 0 ~ 379
Shard 0 contains 379 files
Shard 1 has entries 379 ~ 758
Shard 1 contains 379 files
Shard 2 has entries 758 ~ 1137
Shard 2 contains 379 files
Shard 3 has entries 1137 ~ 1516
Shard 3 contains 379 files
Have 3 entries left over that will be discarded.
[Parallel(n_jobs=4)]: Using backend LokyBackend with 4 concurrent workers.
[Parallel(n_jobs=4)]: Done   4 out of   4 | elapsed:    1.2s finished
Total number of entries in manifest : 1516
Note: we estimated the optimal bucketing duration bins for 10 buckets. You can enable dynamic bucketing by setting the following options in your training script:
  use_lhotse=true
  use_bucketing=true
  num_buckets=10
  bucket_duration_bins=[10.11,12.435,13.3,13.94,14.435,14.835,15.215,15.51,15.865]
  batch_duration=<tune-this-value>
If you'd like to use a different number of buckets, re-estimate this option manually using scripts/speech_recognition/estimate_duration_bins.py
```

Reply:
Perfect